omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierSourceResource.java

```diff
@@ -0,0 +1,303 @@
+            }
+            if (StringUtils.isBlank(url)) { 
+                    errors.add("url"); 
+            }
+            if (!StringUtils.isBlank(username)) { 
+                    identifierSource.setUser(username); 
+            }
+            if (!StringUtils.isBlank(password)) {
+                if(StringUtils.isBlank(username)) {
+                    errors.add("username");
+                }
+                else {
+                    identifierSource.setPassword(password);
+                }
+            }
+            if(errors.toString().length() > 2) {
+                throw new ConversionException("The values of the following inputs are missing or invalid: " + errors.toString());
+            }
+
+            identifierSource.setIdentifierType(patientIdentifierType);
+            identifierSource.setName(name);
+            identifierSource.setUrl(url);
+            identifierSource.setDescription(description);
+
+            savedIdentifierSource =  save(identifierSource);
+
+        }
+        else if(identifierSourceType.equals("IdentifierPool")){
+
+            IdentifierPool identifierSource = new IdentifierPool();
+            String batchSize = validateInput(postBody.get("batchSize"));
+            String minPoolSize = validateInput(postBody.get("minPoolSize"));
+            String sourceUuid = validateInput(postBody.get("sourceUuid"));
+            String sequential = validateInput(postBody.get("sequential"));
+            String refillWithScheduledTask = validateInput(postBody.get("refillWithScheduledTask"));
+            IdentifierSource poolSource = Context.getService(IdentifierSourceService.class).getIdentifierSourceByUuid(sourceUuid);
+
+            if (!StringUtils.isBlank(description)) {
+                identifierSource.setDescription(description);
+            }
+            if (!StringUtils.isBlank(batchSize)) {
+                if(StringUtils.isNumeric(batchSize)) {
+                    identifierSource.setBatchSize(Integer.parseInt(batchSize));
+                }else {
+                    errors.add("batch size");
+                }
+            }
+            if (!StringUtils.isBlank(minPoolSize)) {
+                if(StringUtils.isNumeric(minPoolSize)) {
+                    identifierSource.setMinPoolSize(Integer.parseInt(minPoolSize));
+                    Set<PooledIdentifier> pooledIdentifiers = new LinkedHashSet<PooledIdentifier>();
+                    for(Integer counter = 0; counter < Integer.parseInt(minPoolSize); counter++) {
+                            pooledIdentifiers.add(new PooledIdentifier(identifierSource, "ABC0" + counter));
+                    }
+                    identifierSource.setIdentifiers(pooledIdentifiers);
+                }else {
+                    errors.add("min pool size");
+                }
+            }
+            if (!StringUtils.isBlank(sequential)) {
+                if(sequential.toLowerCase().equals("true") || sequential.toLowerCase().equals("false")) {
```

Comment:
Take a look at this for the values we use for booleans: https://github.com/openmrs/openmrs-module-webservices.rest/blob/master/omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ObsResource1_8.java#L437-L453

Reply:
Created a `validateBoolean()` function from the link to handle boolean inputs.

Reply:
validateABC() usually throws an exception if invalid or returns an error message.